src/renderer/components/node/Node.tsx

```diff
@@ -192,6 +194,11 @@ const NodeInner = memo(({ data, selected }: NodeProps) => {
     const disabled = useDisabled(data);
     const menu = useNodeMenu(data, disabled, { reload: startingNode ? reload : undefined });
 
+    const toggleCollapse = useCallback(() => {
+        setNodeCollapsed(id, !isCollapsed);
+    }, [id, isCollapsed, setNodeCollapsed]);
+    const useCollapse = { isCollapsed: isCollapsed ?? false, toggleCollapse };
```

Comment:
And with this, you effectively disabled memoization for `NodeHeader` and all of its parents...

Why do you use these fake hooks in props in the first place? I never understood that. Why not pass 2 props?

Reply:
Honestly, I have no idea